src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java

```diff
@@ -0,0 +1,136 @@
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.AUTHC_HTTPBASIC_INTERNAL;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class OnBehalfOfJwtAuthenticationTest {
+
+	public static final String POINTER_USERNAME = "/user_name";
+
+	static final TestSecurityConfig.User ADMIN_USER = new TestSecurityConfig.User("admin").roles(ALL_ACCESS);
+
+	public static final String issuer = "cluster_0";
+	public static final String subject = "testUser";
+	public static final String audience = "audience_0";
+	public static final Integer expirySeconds = 100000;
+	public static final String headerName = "Authorization";
+	public static final List<String> roles = List.of("admin", "HR");
+	public static final List<String> backendRoles = List.of("IT");
+
+	private static final String signingKey = Base64.getEncoder().encodeToString("jwt signing key for an on behalf of token authentication backend for testing of OBO authentication".getBytes(StandardCharsets.UTF_8));
+	private static final String encryptionKey = Base64.getEncoder().encodeToString("encryptionKey".getBytes(StandardCharsets.UTF_8));
+
+	private static final OnBehalfOfJwtAuthorizationHeaderFactory tokenFactory = new OnBehalfOfJwtAuthorizationHeaderFactory(
+			signingKey,
+			issuer,
+			subject,
+			audience,
+			roles,
+			backendRoles,
+			expirySeconds,
+			headerName,
+			encryptionKey
+	);
+
+	public static final String ADMIN_USER_NAME = "admin";
+	public static final String DEFAULT_PASSWORD = "secret";
+	public static final String OBO_TOKEN_REASON = "{\"reason\":\"Test generation\"}";
+	public static final String OBO_ENDPOINT_PREFIX = "_plugins/_security/api/user/onbehalfof";
+
+	@ClassRule
+	public static final LocalCluster cluster = new LocalCluster.Builder()
+			.clusterManager(ClusterManager.SINGLENODE).anonymousAuth(false)
+			.users(ADMIN_USER)
+			.nodeSettings(Map.of(
+					"plugins.security.allow_default_init_securityindex", true,
+					"plugins.security.restapi.roles_enabled", List.of("user_admin__all_access")
+			))
+			.authc(AUTHC_HTTPBASIC_INTERNAL)
+			.onBehalfOf(new OnBehalfOfConfig().signing_key(signingKey).encryption_key(encryptionKey))
+			.build();
+
+	@Test
+	public void shouldAuthenticateWithOBOToken() {
+		try(TestRestClient client = cluster.getRestClient(tokenFactory.generateValidToken())){
+
+			TestRestClient.HttpResponse response = client.getAuthInfo();
+
+			response.assertStatusCode(200);
+			String username = response.getTextFromJsonBody(POINTER_USERNAME);
+			assertThat("testUser", equalTo(username));
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Test
+	public void shouldAuthenticateWithOBOTokenEndPoint() {
```

Comment:
What do you think about adding a test that retrieves the token and the tamper with it attempts to use it?

Reply:
I have added another test case~